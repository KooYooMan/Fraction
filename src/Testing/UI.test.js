import React from 'react'
import { mount, render, shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import LecturerHome from '../Lecturer/Home'
import StudentHome from '../Student/MainScreen/Home'
import StudentEnding from '../Student/MainScreen/Ending'
import { Provider } from 'react-redux';
import store from '../store';

configure({ adapter: new Adapter() });

describe('UI debug testing', () => {
    it ('lecturer home screen', () => {
        const component = render(
        <Provider store={store}>
            <LecturerHome />
        </Provider>)
        expect(component).toMatchSnapshot()
    });

    it ('student home screen', () => {
        const component = render(
        <Provider store={store}>
            <StudentHome />
        </Provider>)
        expect(component).toMatchSnapshot()
    });
    
    it ('student ending screen', () => {
        const component = render(
        <Provider store={store}>
            <StudentEnding debug />
        </Provider>)
        expect(component).toMatchSnapshot()
    });
})