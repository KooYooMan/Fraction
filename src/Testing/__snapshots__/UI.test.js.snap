// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI debug testing lecturer home screen 1`] = `
<div
  class="row margin-top"
>
  <div
    class="col-md-4"
  >
    <form
      class="well"
    >
      <h1
        style="font-size:30px"
      >
        Thêm Câu Hỏi
      </h1>
      <br />
      <div
        class="form-group"
      >
        <label
          style="font-size:15px"
        >
          Tử Số
        </label>
        <input
          class="form-control"
          type="number"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          style="font-size:15px"
        >
          Mẫu Số
        </label>
        <input
          class="form-control"
          type="number"
          value=""
        />
      </div>
      <button
        class="btn btn-success"
        type="submit"
      >
        Xác nhận thêm
      </button>
       
    </form>
  </div>
  <div
    class="col-md-8"
  >
    <div
      class="row h35"
    >
      <div
        class="col-md-6"
      >
        <button
          class="btn btn-xs btn-default"
          id="del_rowBtn"
          style="display:none"
        >
          Xóa câu hỏi
        </button>
      </div>
      <div
        class="col-md-offset-2 col-md-4"
      />
    </div>
    <table
      class="table table-hover table-striped table-bordered"
      id="tableSample"
    >
      <thead>
        <tr>
          <th
            style="width:75px"
          >
            Số thứ tự
          </th>
          <th>
            Tử số
          </th>
          <th>
            Mẫu số
          </th>
          <th>
            Tùy chọn
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`UI debug testing student ending screen 1`] = `
<div
  class="contain"
>
  <div
    class="congrats"
  >
    <h1
      class="pa5"
    >
      Chúc mừng
      <span
        class="hide"
      >
        ulation
      </span>
       !
    </h1>
    <div
      class="done"
    >
      <svg
        id="tick"
        style="enable-background:new 0 0 37 37"
        version="1.1"
        viewBox="0 0 37 37"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsxxlink="http://www.w3.org/1999/xlink"
        y="0px"
      >
        <path
          class="circ path"
          d="M30.5,6.5L30.5,6.5c6.6,6.6,6.6,17.4,0,24l0,0c-6.6,6.6-17.4,6.6-24,0l0,0c-6.6-6.6-6.6-17.4,0-24l0,0C13.1-0.2,23.9-0.2,30.5,6.5z"
          style="fill:#0cdcc7;stroke:#07a796;stroke-width:3;stroke-linejoin:round;stroke-miterlimit:10"
        />
        <polyline
          class="tick path"
          points="11.6,20 15.9,24.2 26.4,13.8 "
          style="fill:none;stroke:#fff;stroke-width:3;stroke-linejoin:round;stroke-miterlimit:10"
        />
      </svg>
    </div>
    <div
      class="tc bold text"
    >
      <h1>
        Bạn đã hoàn thành bài tập 
      </h1>
    </div>
    <button
      class="tc ba dib br3 grow bw2 pointer link"
      style="border-color:#02b3e4;color:#02b3e4"
    >
      Quay về màn hình chính
    </button>
  </div>
</div>
`;

exports[`UI debug testing student home screen 1`] = `
<div
  class="center tc items-center"
>
  <h1
    class="pa3 b tc"
  >
     Các bài tập tối giản phân số 
  </h1>
  <div
    class="flex center items-center pa3"
  >
    <button>
      <div
        class="db center tc black link grow ba bw2 b--blue items-center pointer tc"
        title="Tối giản phân số"
      >
        <img
          alt=""
          class="center pa5"
          src="https://dragonlearn.in/assets/card_previews/3389-09a7f07b4baec5efbd2609245b94fc8f.png"
        />
        <dl
          class="lh-copy"
        >
          <dd
            class="ml0"
          >
            Tối giản phân số
          </dd>
        </dl>
      </div>
    </button>
    <button>
      <div
        class="db center tc black link grow ba bw2 b--blue items-center pointer tc"
        title="Tối giản được hoặc không"
      >
        <img
          alt=""
          class="center pa5"
          src="https://dragonlearn.in/assets/card_previews/3390-e5c045b2a558cbda32ecebeeea99a982.png"
        />
        <dl
          class="lh-copy"
        >
          <dd
            class="ml0"
          >
            Tối giản được hoặc không
          </dd>
        </dl>
      </div>
    </button>
    <button>
      <div
        class="db center tc black link grow ba bw2 b--blue items-center pointer tc"
        title="Tối giản từng bước"
      >
        <img
          alt=""
          class="center pa5"
          src="https://dragonlearn.in/assets/card_previews/3391-40aaab2b73a867ce2b37564226a68fd8.png"
        />
        <dl
          class="lh-copy"
        >
          <dd
            class="ml0"
          >
            Tối giản từng bước
          </dd>
        </dl>
      </div>
    </button>
    <button>
      <div
        class="db center tc black link grow ba bw2 b--blue items-center pointer tc"
        title="Tối giản mọi thứ"
      >
        <img
          alt=""
          class="center pa5"
          src="https://dragonlearn.in/assets/card_previews/3392-aa0bb8aa0ba45aecef3a90164cf8f463.png"
        />
        <dl
          class="lh-copy"
        >
          <dd
            class="ml0"
          >
            Tối giản mọi thứ
          </dd>
        </dl>
      </div>
    </button>
    <button>
      <div
        class="db center tc black link grow ba bw2 b--blue items-center pointer tc"
        title="Khi nào thì tối giản"
      >
        <img
          alt=""
          class="center pa5"
          src="https://dragonlearn.in/assets/card_previews/india/hn-49bb6b1269464ab8ee1af3a95aaa3f58.png"
        />
        <dl
          class="lh-copy"
        >
          <dd
            class="ml0"
          >
            Khi nào thì tối giản
          </dd>
        </dl>
      </div>
    </button>
  </div>
  <button>
    <div
      class="tc b--blue ba dib br3 pa3 ma2 grow bw2 shadow-5 pointer center link"
    >
      Quay về màn hình chính
    </div>
  </button>
</div>
`;
